### ISSUE TITLE
Kernel: Implement `madvice`

### ISSUE THREAD

**Issue by Hendiadyoin1** [SerenityOS/serenity#18579]
`madvice` allows a userspace programm to tell the kernel, how it intends to access mapped memory.

Currently we only load mapped files lazily one page at a time, when said page is accessed, which slows down serial IO intensive workloads like multimedia decoders.
For such applications it might be useful to support `MADV_SEQUENTIAL` to increase the loaded pages at a time, or even `MADV_WILLNEED` to start an early (asynchronous) load of the mapped data

Also linux' `MADV_FREE` might be helpful to allow the kernel to for example reclaim parts of the heap that are no longer needed

Additionally `MADV_SEQUENTIAL` can be used as a hint the kernel during memory pressure to prefer early pages of mapped files to evict

[Man page](https://man7.org/linux/man-pages/man2/madvise.2.html)

**Comment by Hendiadyoin1** [comment]
No we don't have any read-ahrad infrastructure yet, it would likely mainly entail creating an async read request and a way of propagating and checking for partial success of it.

This is how I imagine this could work, but might not necessarily be the ideal way of doing it, and definitely is not the easiest/least invasive way,
So take it with a grain of salt

Like when we request to access a page soon, we start a request for the next few pages and don't block on it.
When then accessing a page that is in that mapping, we would look for the request containing it and checking the request for partial completion, ie if that page is already present, or interrogate the corresponding device for such a state, and of that is not the case we should allow waiting for such a partial success.
If we run over the premapped memory we would go through a similar process and just block for the first page, and then let the rest run in the background

**Comment by Hendiadyoin1** [comment]
The FS layer only reads one (fs) block at a time (1 page by "coincidence") but only ever gets requested to do 1 page aswell

**Comment by Panky-codes** [comment]
> The FS layer only reads one (fs) block at a time (1 page by "coincidence") but only ever gets requested to do 1 page aswell

Yeah. But if we have RA in place, FS layer can still request 1 block at a time, but the block layer should send a bigger request (ofc keeping the EOF into account), and have the adjacent blocks in the block cache before the FS does an explicit request for the next block.
